lib/streamlit/runtime/stats.py

```diff
@@ -60,7 +62,8 @@ def marshall_metric_proto(self, metric: MetricProto) -> None:
         metric_point.gauge_value.int_value = self.byte_length
 
 
-class CacheStatsProvider:
+@runtime_checkable
```

Comment:
why is `@runtime_checkable` needed here?

Reply:
Since now `get_stats` is not part of CacheStorage protocol, in cache_data_api.py we check that is implements `CacheStatsProvider` protocol, and only then call it for cache storage: https://github.com/streamlit/streamlit/pull/6052/files#diff-e5462f7904115502e2adeb16b9609d781c6ab56d1b40e6add0a537143c80063eR572